textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/join/JoinDistanceHelper.java

```diff
@@ -0,0 +1,139 @@
+package edu.uci.ics.textdb.dataflow.join;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.lucene.search.MatchAllDocsQuery;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.api.exception.TextDBException;
+import edu.uci.ics.textdb.common.constants.LuceneAnalyzerConstants;
+import edu.uci.ics.textdb.common.constants.DataConstants.KeywordMatchingType;
+import edu.uci.ics.textdb.common.exception.StorageException;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.dataflow.common.JoinDistancePredicate;
+import edu.uci.ics.textdb.dataflow.common.KeywordPredicate;
+import edu.uci.ics.textdb.dataflow.keywordmatch.KeywordMatcherSourceOperator;
+import edu.uci.ics.textdb.storage.DataStore;
+import edu.uci.ics.textdb.storage.relation.RelationManager;
+
+public class JoinDistanceHelper {
+    
+    public static final String BOOK_TABLE_OUTER = "jointest_book_outer";
+    public static final String BOOK_TABLE_INNER = "jointest_book_inner";
+    
+    public static void createTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        
+        // create the two book tables
+        relationManager.createTable(BOOK_TABLE_OUTER, "../index/jointest/book_outer/", 
+                JoinTestConstants.BOOK_SCHEMA, LuceneAnalyzerConstants.standardAnalyzerString());     
+        relationManager.createTable(BOOK_TABLE_INNER, "../index/jointest/book_inner/", 
+                JoinTestConstants.BOOK_SCHEMA, LuceneAnalyzerConstants.standardAnalyzerString());         
+    }
+    
+    public static void insertToOuter(ITuple... tuples) throws StorageException {
```

Comment:
The JoinHelper also provides some helper functions to insert to the two test tables.